docs/execution-providers/Xnnpack-ExecutionProvider.md

```diff
@@ -0,0 +1,87 @@
+---
+title: XNNPACK
+description: Instructions to execute ONNX Runtime with the XNNPACK execution provider
+parent: Execution Providers
+nav_order: 16
+redirect_from: /docs/reference/execution-providers/Xnnpack-ExecutionProvider
+---
+{::options toc_levels="2" /}
+
+# XNNPACK Execution Provider
+
+Accelerate ONNX models on Android devices and WebAssembly with ONNX Runtime and the XNNPACK execution provider. [(XNNPACK)](https://github.com/google/XNNPACK) is a highly optimized library of floating-point neural network inference operators for ARM, WebAssembly, and x86 platforms.
+
+## Contents
+{: .no_toc }
+
+* TOC placeholder
+{:toc}
+
+## Requirements
+
+
+## Install
+Pre-built packages of ONNX Runtime([`onnxruntime-android`](https://mvnrepository.com/artifact/com.microsoft.onnxruntime/onnxruntime-android)) with XNNPACK EP for Android are published on Maven.
+See [here](../install/index.md#install-on-android) for installation instructions.
+
+There is no pre-built package for iOS yet. We will update it once it is ready.
+## Build
+
+Please see the [Build page](../build/eps.md#xnnpack) for instructions on building a package that includes the XNNPACK EP.
+
+Support build for Android/Windows/Linux
+
+## Usage
+
+The ONNX Runtime API details are [here](../api).
+
+The XNNPACK EP can be used via the C, C++ or Java APIs
+
+The XNNPACK EP must be explicitly registered when creating the inference session. For example:
+
+```C++
+Ort::Env env = Ort::Env{ORT_LOGGING_LEVEL_ERROR, "Default"};
+Ort::SessionOptions so;
+so.AppendExecutionProvider("XNNPACK", {{"intra_op_num_threads", std::to_string(intra_op_num_threads)}});
```

Comment:
Is this a required parameter? Or an example of its usage? We should say either way

Reply:
It's an optional parameter, used to set threadpool size for XNNPACK, by provider option.  We explained how to use it in the following sections. And what's the default value if not provided.